src/hypothesis/extra/typehints.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import print_function
+import sys
+from hypothesis import strategies as st
+from hypothesis.strategies import SearchStrategy
+from typing import Dict, Tuple, List, Iterator, Set, Union, Optional, NamedTuple, Callable, Sequence, Iterable, Generator
+import operator
+from collections import OrderedDict
+compose = lambda f,g: lambda *x: f(g(*x))
+# see https://docs.python.org/3/library/typing.html
+try:
+    unicode = unicode
+except:
+    unicode = str
+    from functools import reduce
+
+primitives = {
+    str   : st.text(), # might want to default to `string.printable`
+    int   : st.integers(),
+    bool  : st.booleans(),
+    float : st.floats(),
+    type(None) : st.none(),
+    unicode : st.characters(),
+    bytes : st.binary() # this is weird because str == bytes in py2
+} # missing: fractions, decimal
+
+# it's not possible to make this typesafe, because 
+# we need `issinstance` to narrow down the types of types, 
+# and we are using `issubclass`
+# there's also a FrozenSet type
+# IO types ("file-like")
+# re type
+def type_to_strat(x): # type: (type) -> SearchStrategy
+   '''
+   Given a type, return a strategy which yields a value of that type. Types maybe complex: Union, NamedTuple, etc.
+   For more information, see https://docs.python.org/3/library/typing.html
+   Usage:
+   >> type_to_strat(Union[int,str]).exmample()
+   . . . 3
+   '''
+   if x in primitives:
+       return primitives[x]
+   elif hasattr(x, '_fields'):# NamedTuple isn't a type, it's a function
```

Comment:
Because without this check you can accept `collections.namedtuple` here and fail on access to `_field_types` property later.